thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/components/HoltWintersDetector.java

```diff
@@ -0,0 +1,541 @@
+    // calculate the mean value (= average)
+    double sum = 0.0;
+    for (double num : givenNumbers) {
+      sum += num;
+    }
+    double mean = sum / givenNumbers.size();
+
+    // calculate standard deviation
+    double squaredDifferenceSum = 0.0;
+    for (double num : givenNumbers) {
+      squaredDifferenceSum += (num - mean) * (num - mean);
+    }
+    double variance = squaredDifferenceSum / givenNumbers.size();
+    double standardDeviation = Math.sqrt(variance);
+
+    return zscore * standardDeviation;
+  }
+
+  /**
+   * Fit alpha, beta, gamma by optimizing SSE (Sum of squared errors) using BOBYQA
+   * It is a derivative free bound constrained optimization algorithm
+   * https://en.wikipedia.org/wiki/BOBYQA
+   *
+   * @param y the data
+   * @param lastAlpha last alpha value
+   * @param lastBeta last beta value
+   * @param lastGamma last gamma value
+   * @return double array containing fitted alpha, beta and gamma
+   */
+  private HWParams fitModelWithBOBYQA(double[] y, double lastAlpha, double lastBeta, double lastGamma) {
+    BOBYQAOptimizer optimizer = new BOBYQAOptimizer(7);
+    if (lastAlpha < 0) {
+      lastAlpha = 0.1;
+    }
+    if (lastBeta < 0) {
+      lastBeta = 0.01;
+    }
+    if (lastGamma < 0) {
+      lastGamma = 0.001;
+    }
+    InitialGuess initGuess = new InitialGuess(new double[]{lastAlpha, lastBeta, lastGamma});;
+    MaxIter maxIter = new MaxIter(30000);
+    MaxEval maxEval  = new MaxEval(30000);
+    GoalType goal = GoalType.MINIMIZE;
+    ObjectiveFunction objectiveFunction = new ObjectiveFunction(new MultivariateFunction() {
+      public double value(double[] params) {
+        return forecast(y, params[0], params[1], params[2])[1];
+      }
+    });
+    SimpleBounds bounds = new SimpleBounds(new double[]{0.001, 0.001, 0.001}, new double[]{0.999, 0.999, 0.999});
+
+    HWParams params;
+    try {
+      PointValuePair optimal = optimizer.optimize(objectiveFunction, goal, bounds, initGuess, maxIter, maxEval);
+      params = new HWParams(optimal.getPoint()[0], optimal.getPoint()[1], optimal.getPoint()[2]);
+    } catch (Exception e) {
+      LOG.error(e.toString());
+      params = new HWParams(lastAlpha, lastBeta, lastGamma);
+    }
+    return params;
+  }
+
+  /**
+   * Mapping of sensitivity to zscore on range of 1 - 4
+   * @param sensitivity double from 0 to 10
+   * @return zscore
+   */
+  private static double sensitivityToZscore(double sensitivity) {
+    // If out of bound, use boundary sensitivity
+    if (sensitivity < 0) {
+      sensitivity = 0;
+    } else if (sensitivity > 10) {
+      sensitivity = 10;
+    }
+    double z = 1 + 0.3 * (10 - sensitivity);
+    return z;
+  }
+}
+
+/**
+ * Container class to store holt winters parameters
+ */
+final class HWParams {
+  private final double alpha;
+  private final double beta;
+  private final double gamma;
+
+  public HWParams(double alpha, double beta, double gamma) {
```

Comment:
Use HoltWintersParams or simply ModelParams.
Avoid using abbreviation which is hard to read.